### PR TITLE
Telemetry update

### PR DESCRIPTION
**Pull request by haocs**
The content to be added to Changelog is as follows:
* setting cert thumbprint as the userId while sending telemetry, if the user is using cert based auth for rdfe commands

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@haocs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

File: lib/util/telemetry.js [hunk 1]
```diff
@@ -105,7 +111,8 @@ var _getCurrentSubscription = function() {
     if (sub.user) {
       subscription.user.id = crypto.createHash('sha256').update(sub.user.name).digest('hex');
       subscription.user.type = sub.user.type;
-    } else {
+    } else if (sub.managementCertificate) {
+      subscription.user.id = thumbprint(sub.managementCertificate.cert);
```
Comment (amarzavery) [inline lib/util/telemetry.js]: what if sub.managementCertificate.cert is null or undefined?
check for that and then do the current operations.
If null or undefined then set subscription.user.id to "Not Applicable"

**Comment by azurecla** [comment]
Hi __@haocs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@haocs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by haocs** [comment]
@azurecla please test it

**Pull request closed without merging**